cedar-policy-core/src/entities.rs

```diff
@@ -322,13 +323,13 @@ impl Entities {
 }
 
 /// Create a map from EntityUids to Entities, erroring if there are any duplicates
-fn create_entity_map(es: impl Iterator<Item = Entity>) -> Result<HashMap<EntityUID, Entity>> {
+fn create_entity_map(es: impl Iterator<Item = Entity>) -> Result<HashMap<EntityUID, Arc<Entity>>> {
```

Comment:
should we also have the input `es` here be an iterator of `Arc<Entity>`, and force callers to map `Arc::new` if needed?

Reply:
Ah, I wasn't sure how interested you'd be in propagating this Arc expectation (at least internally). I can make change that reflect this for sure.

Reply:
Updated to this request in 1da976e9668283ae545d252e652b1003f7a682fb, let me know what you think.